src/view/reviewManager.ts

```diff
@@ -73,9 +72,9 @@ export class ReviewManager implements vscode.DecorationProvider {
 				opts.selection = activeTextEditor.selection;
 			}
 
-			vscode.commands.executeCommand('vscode.open', vscode.Uri.file(nodePath.resolve(this._repository.path, params.path)), opts);
+			vscode.commands.executeCommand('vscode.open', vscode.Uri.file(nodePath.resolve(this._repository.rootUri.fsPath, params.path)), opts);
 		}));
-		this._disposables.push(_repository.onDidRunGitStatus(e => {
+		this._disposables.push(_repository.state.onDidChange(e => {
 			// todo, validate state only when state changes.
 			this.updateState();
```

Comment:
we have an existing problem with this that I might write a change for right now (and cause you another merge conflict, sorry), but we really only need to call `updateState` when the repo's HEAD or remotes have changed

Testing these changes made me realize how much of an issue this is :scream

Reply:
> but we really only need to call updateState when the repo's HEAD or remotes have changed

Yeah we can always filter out the events which should not reflect a state update.